packages/valory/connections/openai/connection.py

```diff
@@ -166,21 +171,36 @@ def _get_response(self, prompt_template: str, prompt_values: Dict[str, str]):
         """Get response from openai."""
         # Format the prompt using input variables and prompt_values
         formatted_prompt = prompt_template.format(**prompt_values)
+        engine = self.openai_settings["engine"]
 
         # Call the OpenAI API
-        response = openai.Completion.create(
-            engine=self.openai_settings["engine"],
-            prompt=formatted_prompt,
-            max_tokens=self.openai_settings["max_tokens"],
-            n=1,
-            stop=None,
-            temperature=self.openai_settings["temperature"],
-        )
-
-        # Extract the result from the API response
-        result = response.choices[0].text
+        if engine in ENGINES["chat"]:
+            messages = [
+                {"role": "system", "content": "You are a helpful assistant."},
+                {"role": "user", "content": formatted_prompt},
+            ]
+            response = openai.ChatCompletion.create(
+                model=engine,
+                messages=messages,
+                temperature=self.openai_settings["temperature"],
+                max_tokens=self.openai_settings["max_tokens"],
+                n=1,
+                timeout=120,
+                stop=None,
+            )
+            output = response.choices[0].message.content
```

Comment:
No `timeout` for a single hanging request is given.

Reply:
Can you elaborate on that?

Reply:
You need to set the [request_timeout kwarg](https://github.com/openai/openai-python/blob/e7ee4daaa5b2f4b90c519faee993377ab691559d/openai/api_resources/abstract/engine_api_resource.py#L160) for that. The [timeout](https://github.com/valory-xyz/mech/blob/main/tools/openai_request.py#L64) that you are currently using only takes effect [when retrying](https://github.com/openai/openai-python/blob/e7ee4daaa5b2f4b90c519faee993377ab691559d/openai/api_resources/completion.py#L26-L28).

Reply:
It has already been done.